File: java/src/test/FixtureTest.java
package test;

import static org.junit.Assert.*;

import java.util.List;

import org.junit.Test;

import tablelize.Fixture;
import tablelize.Table;

public class FixtureTest {
	
	
	private class SimpleFixture extends Fixture
	{
		@Override
		public void execute() {
		}

	}

	@Test
	public void testShouldReadExampleFromString() throws Exception {
		
		Fixture fixture = new SimpleFixture();
		
		fixture.loadData("|some|\n|column|\n|50|\n\n");
		
		Table table = fixture.getTable("some");
		
		assertEquals(1, table.rowsCount());
	}
	
	@Test
	public void testMoreThanOneTable() throws Exception {

		Fixture fixture = new SimpleFixture();
		fixture.loadData("|some|\n|column|\n|50|\n\n|other table|\n|other|\n|45|\n\n");	
		
		Table table= fixture.getTable("some");
		Table otherTable= fixture.getTable("other table");

		assertEquals(2, fixture.tablesCount());
		assertEquals(1, table.rowsCount());
		assertEquals(1, otherTable.rowsCount());

	}
	
	@Test
	public void testGetTableShouldThrowIfThereIsMoreThanOneTableWithTheSameName() throws Exception {
		
	      Fixture fixture = new SimpleFixture();
	      fixture.loadData("|TableWithArgs|someArg|\n|field1|field2|\n|1.74|73|\n\n|TableWithArgs|otherArg|\n|field1|field2|\n|474|21|\n\n");

	      try {
	         fixture.getTable("TableWithArgs");
	         fail("should throw an exception here");

	      } catch (Exception e) {}

	}
	
	@Test
	public void testAsAFixtureIWantToIterateOverTablesWithSameName() throws Exception
	{
		Fixture fixture = new SimpleFixture();
		fixture.loadData("|TableWithArgs|someArg|\n|field1|field2|\n|1.74|73|\n\n|TableWithArgs|otherArg|\n|field1|field2|\n|474|21|\n\n");
		
		List<Table> tablesWithSameName = fixture.getTablesWithName("TableWithArgs");
		assertEquals(2, tablesWithSameName.size());

		Table firstTable= tablesWithSameName.get(0);
		Table secondTable= tablesWithSameName.get(1);
		
		assertEquals(1, firstTable.rowsCount());
		assertEquals("1.74", firstTable.getCell("field1", 0));

		assertEquals(1, secondTable.rowsCount());
		assertEquals("474", secondTable.getCell("field1", 0));
	}
	
	@Test
	public void testAsAFixtureIWantToRecoverTablesByTheArgs() throws Exception
	{
		Fixture fixture = new SimpleFixture();
		fixture.loadData("|TableWithArgs|someArg|\n|field1|field2|\n|1.74|73|\n\n|TableWithArgs|otherArg|\n|field1|field2|\n|474|21|\n\n");

		Table tableWithArg= fixture.getTableWithArg("TableWithArgs", 0, "someArg");

		assertEquals(1, tableWithArg.rowsCount());
		assertEquals("1.74", tableWithArg.getCell("field1", 0));

		Table otherTableWithArg= fixture.getTableWithArg("TableWithArgs", 0, "otherArg");

		assertEquals(1, otherTableWithArg.rowsCount());
		assertEquals("474", otherTableWithArg.getCell("field1", 0));
	}

}

File: csharp/TablelizeIt.Test/FixtureTest.cs
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TablelizeIt.Test
{
    class SimpleFixture : Fixture
    {
        public override void Execute() { }
    }

    [TestFixture]
    public class FixtureTest
    {

        [Test]
        public void TestShouldReadExampleFromString()
        {
            SimpleFixture fixture = new SimpleFixture();
            
            fixture.LoadData("|some|\n|column|\n|50|\n\n");
            
            Table table = fixture.GetTable("some");
            
            Assert.AreEqual(1, table.RowsCount());
        }
        
        [Test]
        public void TestMoreThanOneTable() 
        {
            SimpleFixture fixture = new SimpleFixture();
            fixture.LoadData("|some|\n|column|\n|50|\n\n|other table|\n|other|\n|45|\n\n");	
            
            Table table= fixture.GetTable("some");
            Table otherTable= fixture.GetTable("other table");

            Assert.AreEqual(2, fixture.TablesCount());
            Assert.AreEqual(1, table.RowsCount());
            Assert.AreEqual(1, otherTable.RowsCount());
        }
        
        [Test]
        public void TestGetTableShouldThrowIfThereIsMoreThanOneTableWithTheSameName() 
        {
              SimpleFixture fixture = new SimpleFixture();
              fixture.LoadData("|TableWithArgs|someArg|\n|field1|field2|\n|1.74|73|\n\n|TableWithArgs|otherArg|\n|field1|field2|\n|474|21|\n\n");

              try {
                 fixture.GetTable("TableWithArgs");
                 Assert.Fail("should throw an exception here");

              } catch (Exception e) {}

        }
	
        [Test]
        public void TestAsAFixtureIWantToIterateOverTablesWithSameName() 
        {
            SimpleFixture fixture = new SimpleFixture();
            fixture.LoadData("|TableWithArgs|someArg|\n|field1|field2|\n|1.74|73|\n\n|TableWithArgs|otherArg|\n|field1|field2|\n|474|21|\n\n");
            
            List<Table> tablesWithSameName = fixture.GetTablesWithName("TableWithArgs");
            Assert.AreEqual(2, tablesWithSameName.Count);

            Table firstTable= tablesWithSameName[0];
            Table secondTable= tablesWithSameName[1];
            
            Assert.AreEqual(1, firstTable.RowsCount());
            Assert.AreEqual("1.74", firstTable.GetCell("field1", 0));

            Assert.AreEqual(1, secondTable.RowsCount());
            Assert.AreEqual("474", secondTable.GetCell("field1", 0));
        }
	
        [Test]
        public void TestAsAFixtureIWantToRecoverTablesByTheArgs() 
        {
            SimpleFixture fixture = new SimpleFixture();
            fixture.LoadData("|TableWithArgs|someArg|\n|field1|field2|\n|1.74|73|\n\n|TableWithArgs|otherArg|\n|field1|field2|\n|474|21|\n\n");

            Table tableWithArg= fixture.GetTableWithArg("TableWithArgs", 0, "someArg");

            Assert.AreEqual(1, tableWithArg.RowsCount());
            Assert.AreEqual("1.74", tableWithArg.GetCell("field1", 0));

            Table otherTableWithArg= fixture.GetTableWithArg("TableWithArgs", 0, "otherArg");

            Assert.AreEqual(1, otherTableWithArg.RowsCount());
            Assert.AreEqual("474", otherTableWithArg.GetCell("field1", 0));
        }
    }
}
